azurerm/internal/services/apimanagement/api_management_gateway_api_resource_test.go

```diff
@@ -0,0 +1,133 @@
+	r := ApiManagementGatewayAPIResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (ApiManagementGatewayAPIResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := azure.ParseAzureResourceID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroup := id.ResourceGroup
+	serviceName := id.Path["service"]
+	gatewayId := id.Path["gateways"]
+	apiName := id.Path["apis"]
+
+	if resp, err := clients.ApiManagement.GatewayApisClient.GetEntityTag(ctx, resourceGroup, serviceName, gatewayId, apiName); err != nil {
+		if utils.ResponseWasNotFound(resp) {
+			return nil, fmt.Errorf("reading ApiManagement Gateway (%s): %+v", id, err)
+		}
+
+		if !utils.ResponseWasStatusCode(resp, http.StatusNoContent) {
+			return nil, fmt.Errorf("reading ApiManagement Gateway (%s): %+v", id, err)
+		}
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (ApiManagementGatewayAPIResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_api_management" "test" {
+  name                = "acctestAM-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  publisher_name      = "pub1"
+  publisher_email     = "pub1@email.com"
+
+  sku_name = "Developer_1"
+}
+
+resource "azurerm_api_management_gateway" "test" {
+  name              = "acctestAMGateway-%d"
+  api_management_id = azurerm_api_management.test.id
+  description       = "this is a test gateway"
+
+  location_data {
+    name     = "old world"
+    city     = "test city"
+    district = "test district"
+    region   = "test region"
+  }
+}
+
+resource "azurerm_api_management_api" "test" {
+  name                = "acctestapi-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  api_management_name = azurerm_api_management.test.name
+  display_name        = "api1"
+  path                = "api1"
+  protocols           = ["https"]
+  revision            = "1"
+}
+
+resource "azurerm_api_management_gateway_api" "test" {
+  gateway_id = azurerm_api_management_gateway.test.id
+  api_id     = azurerm_api_management_api.test.id
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger)
+}
+
+func (r ApiManagementGatewayAPIResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_api_management_gateway_api" "import" {
+  gateway_id = azurerm_api_management_gateway.test.id
+  api_id     = azurerm_api_management_api.test.id
```

Comment:
Can we use the azurerm_api_management_gateway_api attributes here, to make it more obvious that it's duplicating that resource

```suggestion
  api_management_gateway_id = azurerm_api_management_gateway_api.test.api_management_gateway_id
  api_id                    = azurerm_api_management_gateway_api.test.api_id
```

Reply:
Done